### PR TITLE
Just LINUX things (and friends)

### PR DESCRIPTION
**Pull request by hcorion**
Summary of Changes
- Add option to build using shared llvm libs for Gentoo users, fixes issue #3166 .
- Document some of the build options in README and CMakeLists.txt
- Add warning about needing to install libglvnd for NVIDIA linux users.
- Change MacOSX -> MacOS, fixes one of the issues in #3254 
- Implement progress bar in taskbar for linux.
- Weird TLoU SslCertificateLoader fix, thanks @clienthax for debugging.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellSsl.cpp [hunk 1]
```diff
@@ -69,9 +69,9 @@ s32 cellSslCertificateLoader(u64 flag, vm::ptr<char> buffer, u32 size, vm::ptr<u
 			final.append(getCert(certPath, i));
 		}
 
-		memset(buffer.get_ptr(), 0, size-1);
-		strncpy(buffer.get_ptr(), final.c_str(), size - 1);
-		buffer.get_ptr()[size - 1] = '\0';
+		memset(buffer.get_ptr(), 0, size);
+		strncpy(buffer.get_ptr(), final.c_str(), size);
+		buffer.get_ptr()[size] = '\0';
```
Comment (Farseer2) [inline rpcs3/Emu/Cell/Modules/cellSsl.cpp]: Wait, isn't this guaranteed to write outside the buffer?

Comment (hcorion) [reply rpcs3/Emu/Cell/Modules/cellSsl.cpp]: It doesn't break any games that use it, and TLoU no longer crashes.

Comment (danilaml) [reply rpcs3/Emu/Cell/Modules/cellSsl.cpp]: It's not a proper rationale for introducing a erroneous behaviour/potential UB (though it's is a PS3 buffer that get's written here). It's not even a proper hack.
Unless that's what a real PS3 does it shouldn't be merged into master, especially without a big warning comment. Nobody wants for rpcs3 to turn into a collection of sneaky hacks that will eventually break games as emulation improves.

Comment (hcorion) [reply rpcs3/Emu/Cell/Modules/cellSsl.cpp]: Yes, I agree. I'll write some HW tests soon to confirm the functionality.

Comment (danilaml) [reply rpcs3/Emu/Cell/Modules/cellSsl.cpp]: btw, not sure how helpful the current state of this https://github.com/RPCS3/ps3autotests repo is but there might be a bunch of useful tests (not related to this issue, I think they mainly tests the hw and not sony libs.

**Pull request merged**